lib/sidebar_tree_builder.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: false
+
+require 'forwardable'
+require_relative 'sidebar_page_collection'
+
+module SwedbankPay
+  # Arranges Sidebar pages into a tree
+  class SidebarTreeBuilder
+    include Enumerable
+    extend Forwardable
+    def_delegators :@pages, :each, :length, :empty?, :<<, :[], :count
+
+    def initialize(pages)
+      raise ArgumentError, 'Pages must be a Hash' unless pages.is_a? Hash
+
+      @pages = tree(pages)
+    end
+
+    def to_s
+      stringify
+    end
+
+    def inspect
+      stringify(inspection: true)
+    end
+
+    private
+
+    def stringify(inspection: false)
+      s = "<#{self.class} ##{count}"
```

Comment:
```suggestion
      output_string = "<#{self.class} ##{count}"
```
One letter variables should be avoided 🥰